var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -891,30 +891,38 @@ def determine_version(cls, lib):
             pattern = None
             if ext == "dylib":
                 # Darwin switches the order of the version compared to Linux
-                pattern = re.compile(r"lib(\S*?)_(\S*)\.(\d+\.\d+\.\d+)\.%s" % ext)
+                pattern = re.compile(r"libopencv_(\S*?)\.(\d+\.\d+\.\d+)\.%s" % ext)
             else:
-                pattern = re.compile(r"lib(\S*?)_(\S*)\.%s\.(\d+\.\d+\.\d+)" % ext)
+                pattern = re.compile(r"libopencv_(\S*?)\.%s\.(\d+\.\d+\.\d+)" % ext)
             match = pattern.search(lib)
             if match:
-                ver = match.group(3)
+                ver = match.group(2)
         return ver
```

Comment:
Can you indent this line by 8 spaces? We can return immediately once we detect the version instead of looping over all libraries.

Reply:
done